sstables/sstable_set.hh

```diff
@@ -99,8 +106,63 @@ public:
         selection select(const dht::ring_position_view& pos) const;
     };
     incremental_selector make_incremental_selector() const;
+
+    flat_mutation_reader create_single_key_sstable_reader(
+        column_family*,
+        schema_ptr,
+        reader_permit,
+        utils::estimated_histogram&,
+        const dht::partition_range&, // must be singular
+        const query::partition_slice&,
+        const io_priority_class&,
+        tracing::trace_state_ptr,
+        streamed_mutation::forwarding,
+        mutation_reader::forwarding) const;
+
+    /// Read a range from the sstable set.
+    ///
+    /// The reader is unrestricted, but will account its resource usage on the
+    /// semaphore belonging to the passed-in permit.
+    flat_mutation_reader make_range_sstable_reader(
+        schema_ptr,
+        reader_permit,
+        const dht::partition_range&,
+        const query::partition_slice&,
+        const io_priority_class&,
+        tracing::trace_state_ptr,
+        streamed_mutation::forwarding,
+        mutation_reader::forwarding,
+        read_monitor_generator& rmg = default_read_monitor_generator()) const;
+
+    // Filters out mutations that don't belong to the current shard.
+    flat_mutation_reader make_local_shard_sstable_reader(
```

Comment:
Unrelated: @raphaelsc I thought we don't need this guy anymore now that resharding happens on startup. If there is a single user (resharding compaction) maybe we could fold this into it, instead of increasing the number of overloads here.

Reply:
@denesb even resharding compaction is not interested in this local shard sst reader. resharding gets data from all shards and segregate it using special mutation writer. so we can get rid of make_local_shard_sstable_reader(). i'll eventually clean it up if nobody do it first. we have a few more cleanup work to do since after we changed resharding.